### ISSUE TITLE
failed to start child: Quantum.TaskStagesSupervisor and Quantum.ExecutorSupervisor

### ISSUE THREAD

**Issue by tanweerdev** [c-rack/quantum-elixir#299]
**Config** contains

```elixir
config :qber, Qber.Scheduler,
  jobs: [
    {"* * * * *", {IO, :inspect, [:beep]}}
    # Every minute
    # {"* * * * *",      {Heartbeat, :send, []}}
  ]
```

```elixir
Application contains
children = [
      supervisor(Qber.Repo, []),
      supervisor(Qber.Endpoint, []),
      worker(Qber.Scheduler, []),
      supervisor(Exq, [])
    ]

   opts = [strategy: :one_for_one, name: Qber.Supervisor]
   Supervisor.start_link(children, opts)
```

and my **Scheduler contains** 

```elixir
defmodule Qber.Scheduler do
  @moduledoc """

  """
  use Quantum.Scheduler, otp_app: :qber
end
```

After adding these **quantum configs** I got the following error

```
** (Mix) Could not start application qber: Qber.start(:normal, []) returned an error: shutdown: failed to start child: Qber.Scheduler
    ** (EXIT) shutdown: failed to start child: Quantum.TaskStagesSupervisor
        ** (EXIT) shutdown: failed to start child: Quantum.ExecutorSupervisor
            ** (EXIT) an exception was raised:
                ** (FunctionClauseError) no function clause matching in ConsumerSupervisor.init/2
                    (gen_stage) lib/consumer_supervisor.ex:265: ConsumerSupervisor.init([{Quantum.Executor, {Qber.Scheduler.Task.Supervisor, Qber.Scheduler.TaskRegistry}}], [strategy: :one_for_one, subscribe_to: [{Qber.Scheduler.ExecutionBroadcaster, [max_demand: 50]}]])
                    (gen_stage) lib/consumer_supervisor.ex:287: ConsumerSupervisor.init/1
                    (gen_stage) lib/gen_stage.ex:1825: GenStage.init/1
                    (stdlib) gen_server.erl:365: :gen_server.init_it/2
                    (stdlib) gen_server.erl:333: :gen_server.init_it/6
                    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

**Comment by c-rack** [comment]
@tanweerdev the job schedule definition is invalid and should be `"* * * * *"`. If that does not solve the problem, could you please provide a minimal test project (for example a public repo with a test project) that produces the error?

**Comment by tanweerdev** [comment]
@c-rack here is the link to `repo` https://github.com/tanweerdev/quantum_demo. Actually the problem is incompatibility with `gen_stage` old version.

**Comment by maennchen** [comment]
@tanweerdev You're didn't install `timex` (See `README`). Does that solve it?

**Comment by maennchen** [comment]
@tanweerdev Never mind, that does not solve the problem.

@c-rack This was probably introduced by #294. We can now either revert the PR or require `gen_stage >= 0.13`.

**Comment by tanweerdev** [comment]
@maennchen `Pigeon` requires `old gen_server` and thats why I am facing this compatibility issue.

**Comment by maennchen** [comment]
@tanweerdev The library was compliant with `gen_stage:v0.12.2` (I even made some contributions for this version.)

The PR #294 changed how we call the `init` function to get rid of a warning. But the minimal required version was never raised to `~> 0.13` which is required by now.

We can now either revert that change or raise the minimal version. I don't really mind which way we go.

**Comment by tanweerdev** [comment]
@maennchen let me create an issue in `pigoen` library. Btw, `Quantum` should also support `0.12.2` as this is just a minor version change. So it would be better just to generate a warning

**Comment by maennchen** [comment]
@tanweerdev I opened a PR #300

**Comment by tanweerdev** [comment]
@maennchen thanks

**Comment by maennchen** [comment]
Will be released with #301
